modal_proto/api.proto

```diff
@@ -551,9 +551,18 @@ message ContainerArguments {  // This is used to pass data from the worker to th
   optional string checkpoint_id = 14;
 }
 
+message CancelInputEvent {
+  repeated string input_ids = 1;
+}
+
+message ContainerHeartbeatResponse {
+  optional CancelInputEvent cancel_input_event = 1;
+}
+
 message ContainerHeartbeatRequest {
   string current_input_id = 1;
   double current_input_started_at = 2;
+  bool supports_graceful_input_cancellation = 3; // implicitly also supports long poll
```

Comment:
j/w but is the idea that we need this so that we can tell whether the request came from a sufficiently new client? would we be able to remove it at some point in the future?

Reply:
Yeah exactly. There is a change in the responder API for this call that changes it to a long-polling mechanism when this flag is set to true, and in that case it can also return a payload with some data that the older clients wouldn't listen to. When older client versions are no longer supported, we can get rid of this. (Reminds me, I should bump the minor version of the package here too to make the deprecation path clear)